source/internal/_modify.js

```diff
@@ -0,0 +1,29 @@
+import _isArray from './_isArray.js';
+import _isInteger from './_isInteger.js';
+
+/**
+ * Makes a shallow clone of an object, applying the given fn to the specified
+ * property with the given value. Note that this copies and flattens prototype
+ * properties onto the new object as well. All non-primitive properties are
+ * copied by reference.
+ *
+ * @private
+ * @param {String|Number} prop The property name to set
+ * @param {Function} fn The function to apply to the property
+ * @param {Object|Array} obj The object to clone
+ * @return {Object|Array} A new object equivalent to the original except for the changed property.
+ */
+export default function _modify(prop, fn, obj) {
+  if (_isInteger(prop) && _isArray(obj)) {
+    var arr = [].concat(obj);
+    arr[prop] = fn(arr[prop]);
+    return arr;
+  }
+
+  var result = {};
+  for (var p in obj) {
+    result[p] = obj[p];
+  }
```

Comment:
Unless intentional, 
I'd suggest to wrap the assignment into a `hasOwnProperty` 
to avoid performing the operation in the prototype chain

https://steemit.com/programming/@heejin/javascript-hasownproperty-in-for-in-loops

```ts
var obj = {
  a: 1,
  b: 2
};

Object.prototype.haha = 3;

for (var k in obj) {
  console.log(k); // prints a, b, haha
}
```

Reply:
Thank you Giuseppe for your suggestion. I am aware that many linters advise adding own property checks in for loops, however in this day and age (and in this paradigm) I am unsure what is the benefit of this. Prototypal inheritance is wildly discouraged, and does not really fit into the functional paradigm Ramda promotes. I am uncertain about the impact on the performance, but I'm guessing it slows down the loop considerably.

Other than this, the `_modify` function is a pound for pound copy of the `_assoc` function, so I think this question might be beyond the scope of this change. I looked through some similar functions (e. g. `evolve`), and none of them tested for inherited properties. Probably @CrossEye can help whether we should look into it or not.

Reply:
Note that `assoc` already works this way.  From [the docs][as]:

> Note that this copies and flattens prototype properties onto the new object as well. 

So I would stick with that for `modify`.

  [as]: https://ramdajs.com/docs/#assoc

Reply:
agreed, let's follow `assoc`'s implementation